### PR TITLE
[DBZ-1031] Support for tables with reserved names

### PR DESCRIPTION
**Pull request by wscheep**

### CODE REVIEW

File: debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SnapshotIT.java [hunk 1]
```diff
@@ -187,6 +187,22 @@ public void takeSchemaOnlySnapshotAndStartStreaming() throws Exception {
...
+        );
+
+        // Populate database
+        for (int i = 0; i < INITIAL_RECORDS_PER_TABLE; i++) {
+            connection.execute(
+                    String.format("INSERT INTO [User] VALUES(%s, '%s', %s, '%s')", i, "name" + i, new BigDecimal(i + ".23"), "2018-07-18 13:28:56")
+            );
+        }
+
+        TestHelper.enableTableCdc(connection, "User");
```
[the remark this answers is not in the document]
Comment (wscheep) [reply debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SnapshotIT.java]: sure, makes sense, I forgot. 
Does it also make sense to refactor takeSnapshot() to pass a table name? Or do you prefer I duplicate all assertion code?

File: debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java [hunk 2]
```diff
@@ -114,7 +114,8 @@ else if (connectorConfig.getSnapshotLockingMode() == SnapshotLockingMode.EXCLUSI
 
                     LOGGER.info("Locking table {}", tableId);
 
-                    statement.executeQuery("SELECT * FROM " + tableId.table() + " WITH (TABLOCKX)").close();
+                    String query = String.format("SELECT * FROM [%s] WITH (TABLOCKX)", tableId.table());
```
[the remark this answers is not in the document]
Comment (wscheep) [reply debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerSnapshotChangeEventSource.java]: I added it to be consistent with the other select statement in the same class.

File: debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SnapshotIT.java [hunk 3]
```diff
@@ -187,6 +187,22 @@ public void takeSchemaOnlySnapshotAndStartStreaming() throws Exception {
         testStreaming();
     }
 
+    @Test
```
[the remark this answers is not in the document]
Comment (wscheep) [reply debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SnapshotIT.java]: done

File: debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SnapshotIT.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (wscheep) [reply debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SnapshotIT.java]: done

**Comment by gunnarmorling** [comment]
Rebased and applied. Thanks a lot, @wscheep!

**Pull request closed without merging**